### PR TITLE
#167837395 Build endpoint to register and login user to database

### PR DESCRIPTION
**Pull request by danielotieno**
#### What does this PR do?
Signup and Login User API endpoint

#### Description of Task to be completed?
| EndPoint                    | Functionality                  |
| -----------------------------| ------------------------------- |
| POST:  /api/v2/auth/signup      | Register a new User   |
| POST:  /api/v2/auth/signup      | Login regustered User  |

#### How should this be manually tested?

-  Fork the repo here [fork me](https://github.com/danielotieno/way-farer-api)
-  `https://github.com/xxxx/way-farer-api ` **xxxx** is your GitHub username
-  cd `way-farer-api`
-  `yarn install`
-  `yarn run test`
- `yarn start`

Send a `POST` request to `/api/v2/auth/signup` with json payload 

#### What are the relevant pivotal tracker stories?
PT Link: [#167837395](https://www.pivotaltracker.com/story/show/167837395)

### CODE REVIEW

File: server/services/UserService.js [hunk 1]
```diff
@@ -5,20 +5,26 @@ import config from '../config/config'
...
-      first_name: getUser.firstName,
-      last_name: getUser.lastName,
-      email: getUser.email,
+    try {
+      const existingUser = await UserModel.getUserByEmail(req.body.email)
+      if (existingUser)
+        return res
+          .status(400)
+          .send({ status: 400, error: 'User already exists' })
+      req.body.password = EncryptData.generateHash(req.body.password)
```
Comment (njeri-ngigi) [inline server/services/UserService.js]: make use of object destructuring.

```
const { email, password, role, firstName: first_name, lastName: last_name } = req.body;
```

File: server/services/UserService.js [hunk 2]
```diff
@@ -5,20 +5,26 @@ import config from '../config/config'
...
+      req.body.role = 'user'
+      await UserModel.createUser(req.body)
+      const user = {
+        first_name: req.body.firstName,
+        last_name: req.body.lastName,
+        email: req.body.email,
```
Comment (njeri-ngigi) [inline server/services/UserService.js]: After making use of destructuring you could create a new object by
```
const user = { first_name, last_name, email }
```

File: server/services/UserService.js [hunk 3]
```diff
@@ -39,8 +45,8 @@ class UserService {
     )
     const loggedInUser = {
       token,
-      first_name: req.user.firstName,
-      last_name: req.user.lastName,
+      first_name: req.user.first_name,
+      last_name: req.user.last_name,
```
Comment (njeri-ngigi) [inline server/services/UserService.js]: Make use of object destructuring

Comment (danielotieno) [reply server/services/UserService.js]: Thank you, for the feedback 

I am working on it

File: server/services/UserService.js (hunk 2 above)
Comment (danielotieno) [reply server/services/UserService.js]: Awesome 👍

Comment (danielotieno) [reply server/services/UserService.js]: Awesome 👍

**Review by njeri-ngigi — APPROVED** [review]

**Pull request merged**